lib/Models/RegionMapping.js

```diff
@@ -639,22 +645,95 @@ function addDescriptionAndProperties(regionMapping, regionIndices, regionImagery
 
 }
 
+function calculateImageryLayerIntervals(timeColumn, currentTime, animateForwards) {
+    // Before and after are defined according to animation direction
+    var isBefore = animateForwards ? JulianDate.lessThan : JulianDate.greaterThan;
+    var isAfter = animateForwards ? JulianDate.greaterThan : JulianDate.lessThan;
+    // Loop over time intervals and generate a time interval for the current and next imagery layer
+
+    // Need to keep track of 3 times:
+    // - past: The closest transition time in the past (for current interval)
+    // - future1: The closest transition time in the future (for current & next intervals)
+    // - future2: The next closest transition time in the future (for next interval)
+    var past = null;
```

Comment:
Since you're using for / of, lambdas, and other fancy features in this function, you might as well use `const` and `let` too right?

Reply:
I guess I just followed what the rest of the file was using for variable assignment. I tend to follow `require` vs `import` and `let`/`const` vs `var`, but interchange `function() { }` with `() => {}`, though I do agree it would make more sense to use `const` and `let`. (In this specific case block scope = function scope, so there's no advantage with `let`)

Reply:
yep that makes sense, I don't have any problem with that.

I guess our long term goal should be to move our code toward newer ES features; we just don't want to create a mixed mess in the meantime.  So with that in mind, I think it's fine to write new functions with newer style even if the rest of the file uses an older style.  Also upgrading a whole function is fine.  And if you're really ambitious, upgrading the whole file is great.

If you're making major-but-trivial changes to existing code, though, it'd be good to keep those changes separate for sanity of PR review.  So say you want to replace var with const/let in a whole file.  Ideally you'd do it in a separate branch, PR it, insta-merge it, and then pull it into the branch you were originally working in to make further changes.

TL;DR: you don't need to do anything.  I'm just laying out how I think we should approach this in general.